### DISCUSSION TITLE
[VS2022] Copilot adds trailing whitespaces for every new line

### DISCUSSION THREAD

**Discussion by LMSDev** [community/community#14115] [Copilot Conversations] [answered]
Whenever the GitHub Copilot extension is enabled in my VS 2022 C# environment, I get trailing whitespaces added for each newline to match the current indentation. If I disable the extension, the problem stops. It does not matter whether Copilot actually suggests something for the new line or not, it always happens. See screenshot below of when I just hit return and spaces are added automatically. The expected behavior would be no spaces.

![image](https://user-images.githubusercontent.com/10090087/161398860-44064323-8028-47b8-8a20-ef6bbe5ec23c.png)

**Comment by kzu** [comment, marked as answer]
Yes, this is consistent with the way it works in VS Code, and is the only way Copilot can properly format de-indent suggestions. Visual Studio provides this concept of a "virtual whitespace" (which you're referring to with 'no spaces' even though the caret is not at the beginning of the line) that has no equivalent in the "text-based world" that Copilot lives in, and so this trade-off had to be put in place. 

Perhaps you can turn off "show whitespace"?

**Reply by LMSDev** [reply]
Thanks for clearing that up! I only noticed it because the "useless" whitespaces were caught by a code style plugin (hence the curly lines), so I will just disable that rule.
